### PR TITLE
fix: Configure unix socket library for Graalvm

### LINKED ISSUE #1940: support jdbc connection with unix domain socket in native image
### Feature Description

I'm sorry I don't really know where this issue should be submit, so I'll submit it here first, and let me know if it shouldn't be.

I've looked at the [previous archived repositories](https://github.com/GoogleCloudPlatform/native-image-support-java), but I don't know what repositories the previous functionality has been moved to.

I've submitted a PR to that repository before: https://github.com/GoogleCloudPlatform/native-image-support-java/pull/304

That PR add unix domain socket supporting in graalvm native images.

But now when I add dependency according to the new documentation, I still get the same error as before this PR

```
java.lang.ClassNotFoundException: jnr.ffi.provider.jffi.Provider
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:123) ~[na:na]
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:87) ~[na:na]
        at java.base@17.0.10/java.lang.Class.forName(DynamicHub.java:1322) ~[report-service:na]
        at java.base@17.0.10/java.lang.Class.forName(DynamicHub.java:1285) ~[report-service:na]
        at java.base@17.0.10/java.lang.Class.forName(DynamicHub.java:1278) ~[report-service:na]
        at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.getInstance(FFIProvider.java:68) ~[na:na]
        at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.<clinit>(FFIProvider.java:57) ~[na:na]
        at jnr.ffi.provider.FFIProvider.getSystemProvider(FFIProvider.java:35) ~[report-service:na]
        at jnr.ffi.Runtime$SingletonHolder.<clinit>(Runtime.java:109) ~[na:na]
        at jnr.ffi.Runtime.getSystemRuntime(Runtime.java:70) ~[report-service:na]
        at jnr.unixsocket.SockAddrUnix.<init>(SockAddrUnix.java:46) ~[report-service:na]
        at jnr.unixsocket.SockAddrUnix$DefaultSockAddrUnix.<init>(SockAddrUnix.java:208) ~[na:na]
        at jnr.unixsocket.SockAddrUnix.create(SockAddrUnix.java:174) ~[report-service:na]
        at jnr.unixsocket.UnixSocketAddress.<init>(UnixSocketAddress.java:47) ~[na:na]
        at com.google.cloud.sql.core.Connector.connect(Connector.java:106) ~[na:na]
        at com.google.cloud.sql.core.InternalConnectorRegistry.connect(InternalConnectorRegistry.java:179) ~[na:na]
        at com.google.cloud.sql.mysql.SocketFactory.connect(SocketFactory.java:63) ~[report-service:1.16.0]
        at com.google.cloud.sql.mysql.SocketFactory.connect(SocketFactory.java:45) ~[report-service:1.16.0]
```

I think that repository was using an older version of the code before moving it to another repository, so the code in the last batch of PRs didn't get moved to the new repository.

If this functionality should be moved to this repository, is it possible to merge the code from the previous PR to this repository?

### Sample code

_No response_

### Alternatives Considered

_No response_

### Additional Details

_No response_

### PR DESCRIPTION
**Pull request by hessjcg**
This is a fix for the graalvm configuration issues related to the jnr.unixsocket library. 

- The proxy classes are now being properly configured for reflection and dynamic proxy.
- Native library files that are usually loaded as resources from jffi-1.3.13-native.jar are included in the build. 

Fixes #1940

### CODE REVIEW

**Review by enocom — COMMENTED** [review]
Is this a "partial" fix of the linked issue? Or a complete fix?

**Comment by hessjcg** [comment]
This is a complete fix. I've updated the comment.

**Review by jackwotherspoon — APPROVED** [review]

**Comment by enocom** [comment]
cc @mpeddada1 as FYI

File: core/src/main/resources/META-INF/native-image/com.google.cloud.sql/cloud-sql-jdbc-socket-factory-parent/native-image.properties [hunk 1]
```diff
@@ -1,3 +1,8 @@
 Args =\
+  -H:+UnlockExperimentalVMOptions \
   -H:+AddAllCharsets \
+  -H:+BuildReport \
+  -H:ReflectionConfigurationResources=META-INF/native-image/com.google.cloud.sql/cloud-sql-jdbc-socket-factory-parent/jni-unix-socket-config.json \
+  -H:ResourceConfigurationResources=META-INF/native-image/com.google.cloud.sql/cloud-sql-jdbc-socket-factory-parent/resource-config.json \
+  -H:DynamicProxyConfigurationResources=META-INF/native-image/com.google.cloud.sql/cloud-sql-jdbc-socket-factory-parent/proxy-config.json \
```
Comment (mpeddada1) [inline core/src/main/resources/META-INF/native-image/com.google.cloud.sql/cloud-sql-jdbc-socket-factory-parent/native-image.properties:7]: optional q: I wonder if we need to specify these JSONs in native-image.properties? The compiler automatically picks up config files under the `META-INF/native-image` directory if the files follow a standard naming pattern ([link](https://www.graalvm.org/latest/reference-manual/native-image/overview/BuildConfiguration/#embed-a-configuration-file))

**Pull request merged**